### ISSUE TITLE
Error(s) in loading state_dict for ResnetGenerator:

### ISSUE THREAD

**Issue by HareshMiriyala** [junyanz/pytorch-CycleGAN-and-pix2pix#671]
Using PyTorch 1.1.0 , Python3 (non conda version)

It throws the following error : 

Error(s) in loading state_dict for ResnetGenerator:
	Missing key(s) in state_dict: "model.10.conv_block.5.weight", "model.10.conv_block.6.running_mean", "model.10.conv_block.6.running_var", "model.10.conv_block.6.bias", "model.11.conv_block.5.weight", "model.11.conv_block.6.running_mean", "model.11.conv_block.6.running_var", "model.11.conv_block.6.bias", "model.12.conv_block.5.weight", "model.12.conv_block.6.running_mean", "model.12.conv_block.6.running_var", "model.12.conv_block.6.bias", "model.13.conv_block.5.weight", "model.13.conv_block.6.running_mean", "model.13.conv_block.6.running_var", "model.13.conv_block.6.bias", "model.14.conv_block.5.weight", "model.14.conv_block.6.running_mean", "model.14.conv_block.6.running_var", "model.14.conv_block.6.bias", "model.15.conv_block.5.weight", "model.15.conv_block.6.running_mean", "model.15.conv_block.6.running_var", "model.15.conv_block.6.bias", "model.16.conv_block.5.weight", "model.16.conv_block.6.running_mean", "model.16.conv_block.6.running_var", "model.16.conv_block.6.bias", "model.17.conv_block.5.weight", "model.17.conv_block.6.running_mean", "model.17.conv_block.6.running_var", "model.17.conv_block.6.bias", "model.18.conv_block.5.weight", "model.18.conv_block.6.running_mean", "model.18.conv_block.6.running_var", "model.18.conv_block.6.bias". 
	Unexpected key(s) in state_dict: "model.10.conv_block.7.weight", "model.10.conv_block.7.bias", "model.10.conv_block.7.running_mean", "model.10.conv_block.7.running_var", "model.10.conv_block.7.num_batches_tracked", "model.11.conv_block.7.weight", "model.11.conv_block.7.bias", "model.11.conv_block.7.running_mean", "model.11.conv_block.7.running_var", "model.11.conv_block.7.num_batches_tracked", "model.12.conv_block.7.weight", "model.12.conv_block.7.bias", "model.12.conv_block.7.running_mean", "model.12.conv_block.7.running_var", "model.12.conv_block.7.num_batches_tracked", "model.13.conv_block.7.weight", "model.13.conv_block.7.bias", "model.13.conv_block.7.running_mean", "model.13.conv_block.7.running_var", "model.13.conv_block.7.num_batches_tracked", "model.14.conv_block.7.weight", "model.14.conv_block.7.bias", "model.14.conv_block.7.running_mean", "model.14.conv_block.7.running_var", "model.14.conv_block.7.num_batches_tracked", "model.15.conv_block.7.weight", "model.15.conv_block.7.bias", "model.15.conv_block.7.running_mean", "model.15.conv_block.7.running_var", "model.15.conv_block.7.num_batches_tracked", "model.16.conv_block.7.weight", "model.16.conv_block.7.bias", "model.16.conv_block.7.running_mean", "model.16.conv_block.7.running_var", "model.16.conv_block.7.num_batches_tracked", "model.17.conv_block.7.weight", "model.17.conv_block.7.bias", "model.17.conv_block.7.running_mean", "model.17.conv_block.7.running_var", "model.17.conv_block.7.num_batches_tracked", "model.18.conv_block.7.weight", "model.18.conv_block.7.bias", "model.18.conv_block.7.running_mean", "model.18.conv_block.7.running_var", "model.18.conv_block.7.num_batches_tracked". 
	size mismatch for model.10.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.11.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.12.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.13.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.14.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.15.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.16.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.17.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).
	size mismatch for model.18.conv_block.6.weight: copying a param with shape torch.Size([256, 256, 3, 3]) from checkpoint, the shape in current model is torch.Size([256]).

**Comment by junyanz** [comment]
Could you share with more details (e.g., training and test command)?

**Comment by HareshMiriyala** [comment]
Hey, I'm closing this issue for now, for anyone encountering similar error with pix2pix, I resolved it by setting dropout as True. It works with PyTorch 1.1.0.

**Comment by zhhezhhe** [comment]
I meet the same error. It seems that there are dropout layers when training but no dropout layers when test.
(10): ResnetBlock(
        (conv_block): Sequential(
          (0): ReflectionPad2d((1, 1, 1, 1))
          (1): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), bias=False)
          (2): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (3): ReLU(inplace=True)
          (4): Dropout(p=0.5, inplace=False)
          (5): ReflectionPad2d((1, 1, 1, 1))
          (6): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), bias=False)
          (7): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
(10): ResnetBlock(
        (conv_block): Sequential(
          (0): ReflectionPad2d((1, 1, 1, 1))
          (1): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), bias=False)
          (2): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (3): ReLU(inplace=True)
          (4): ReflectionPad2d((1, 1, 1, 1))
          (5): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), bias=False)
          (6): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
